pvanalytics/features/daylight.py

```diff
@@ -0,0 +1,97 @@
+"""Functions for identifying when the sun is up."""
+import numpy as np
+import pandas as pd
+
+
+def _minute_of_day(time):
+    # return the minute of the day (counting from midnight).
+    return time.hour * 60 + time.minute
+
+
+def frequency(power_or_irradiance, threshold=0.8, minimum_days=60):
+    """Identify daytime periods based on frequency of positive data.
+
+    Data is aggregated by minute of the day and the mean number of
+    positive values for each minute is calculated. Each minute with at
+    least `threshold` times the mean number of positive values is
+    considered day-time.
+
+    Parameters
+    ----------
+    power_or_irradiance : Series
+        DatetimeIndexed series of power or irradiance measurements.
+    threshold : float, default 0.8
+        Fraction of data that must have positive power or irradiance
+        measurements for a time to be considered "day".
+    minimum_days : int, default 60
+        Minimum number of days with data. If `power_or_irradiance` has
+        fewer days with positive data then a ValueError is raised.
+
+    Returns
+    -------
+    Series
+        A DatetimeIndexed series with True for each index that is
+        during daylight hours.
+
+    Raises
+    ------
+    ValueError
+        if there are less than `minimum_days` of positive data.
+
+    Notes
+    -----
+    This function is derived from the pvfleets_qa_analysis
+    project. Copyright (c) 2020 Alliance for Susteainable Energy, LLC.
+
+    """
+    # remove all datapoints less than or equal to zero
+    data = power_or_irradiance.copy()
+    data[data <= 0] = np.nan
+    minutes = pd.Series(_minute_of_day(data.index), index=data.index)
+    # group by minute of day and compute frequency of positive values
+    # at each minute
+    value_frequency = data.groupby(minutes).count()
+    if value_frequency.max() < minimum_days:
+        raise ValueError("Too few days with data (got {}, minimum_days={})"
+                         .format(value_frequency.max(), minimum_days))
+    daylight_minutes = value_frequency[
+        value_frequency > threshold * value_frequency.mean()
```

Comment:
@matt14muller I don't understand how this works. `value_frequency[minute_of_day]` is the count of days when the data value is positive. `value_frequency.mean()` is the average of these counts. What happens if the data span a summer a high latitude? Since the daylight period is long, `value_frequency.mean()` will be large compared to data from a winter at the same latitude. So it is more likely to label a minute as daytime during winter than summer?